$NetBSD$

Allow variables to be overridden and integrated into a build system. 

--- Makefile.orig	2016-12-04 03:42:47.000000000 +0000
+++ Makefile
@@ -13,8 +13,9 @@ SRC = src
 CXXFLAGS = -I$(SRC) -std=c++11 -MD $(OPTDEBUGFLAGS) -Wall -Wshadow -Wsign-compare -Werror
 LIBS = -lm
 
-DESTDIR = /usr/local
-ICEBOX = /usr/local/share/icebox
+DESTDIR ?= /usr/local
+ICEBOX ?= /usr/local/share/icebox
+PREFIX ?=
 
 .PHONY: all
 all: bin/arachne-pnr share/arachne-pnr/chipdb-1k.bin share/arachne-pnr/chipdb-8k.bin
@@ -104,16 +105,16 @@ mxebin:
 
 .PHONY: install
 install: all
-	mkdir -p $(DESTDIR)/bin
-	cp bin/arachne-pnr $(DESTDIR)/bin/arachne-pnr
-	mkdir -p $(DESTDIR)/share/arachne-pnr
-	cp share/arachne-pnr/chipdb-1k.bin $(DESTDIR)/share/arachne-pnr/chipdb-1k.bin
-	cp share/arachne-pnr/chipdb-8k.bin $(DESTDIR)/share/arachne-pnr/chipdb-8k.bin
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	cp bin/arachne-pnr $(DESTDIR)$(PREFIX)/bin/arachne-pnr
+	mkdir -p $(DESTDIR)$(PREFIX)/share/arachne-pnr
+	cp share/arachne-pnr/chipdb-1k.bin $(DESTDIR)$(PREFIX)/share/arachne-pnr/chipdb-1k.bin
+	cp share/arachne-pnr/chipdb-8k.bin $(DESTDIR)$(PREFIX)/share/arachne-pnr/chipdb-8k.bin
 
 .PHONY: uninstall
 uninstall:
-	rm -f $(DESTDIR)/bin/arachne-pnr
-	rm -f $(DESTDIR)/bin/share/arachne-pnr/*.bin
+	rm -f $(DESTDIR)$(PREFIX)/bin/arachne-pnr
+	rm -f $(DESTDIR)$(PREFIX)/bin/share/arachne-pnr/*.bin
 
 .PHONY: clean
 clean:
